### DISCUSSION TITLE
Folder Read Order

### DISCUSSION THREAD

**Discussion by ScottCov** [dadoonet/fscrawler#1666] [General]
I have a few hundred folders with Pdfs. What determines the order fscrawler works thru the folders? For instance, when I start indexing for the first time or reindex using --restart, It always begins with folder 20-34966, even though this isn't the highest/lowest sequentially nor is it the earliest or latest date?

![image](https://github.com/dadoonet/fscrawler/assets/119634746/b34be1ed-78f8-45d0-b959-a1b78a7ee739)

**Comment by dadoonet** [comment]
I have no idea.

It's the way Java is [returning the files](https://github.com/dadoonet/fscrawler/blob/master/crawler/crawler-fs/src/main/java/fr/pilato/elasticsearch/crawler/fs/crawler/fs/FileAbstractorFile.java#L92-L95).

```java
File[] files = new File(dir).listFiles(file -> {
    if (fsSettings.getFs().isFollowSymlinks()) return true;
    return !Files.isSymbolicLink(file.toPath());
});
``` 

But that could be a nice addition to fetch the content ordered by the newest content first...
Would you like to open an issue?

**Reply by ScottCov** [reply]
It was more a matter of curiosity
